ptd_client_server/lib/server.py

```diff
@@ -875,6 +876,22 @@ def stop(self, mode: Mode) -> bool:
                     channels_amount,
                 )
 
+                # we will generate crude max power approximation and depending on that, we will add fix to crest factor
+                # default is crest factor 3 (pek current is 3x rms current)
+                # PSUs under 75W don't have mandatory Power Factor Correction, so they can be arbitrarily dirty
+                # meters use crest factor of 6 for such devices. so, most meaningful solution is to increase range x2
+                # that will increase detectable peak current to 6x original max noticed value
+                # in case such things are not done, peaks over 3x range will be cut off, so measured/reported power will be reported as lower than realistic
+
+                if self._maxAmps is not None and self._maxVolts is not None:
+                    w = float(self._maxAmps) * float(self._maxVolts)
```

Comment:
Here power factor must also be multiplied right?

Reply:
exactly. and to get that, i need to modify max_volts_amps which will break all unit tests and when doing that, i'll just either base decision on power factor alone or rather on real power

Reply:
Oh sure. I think using power factor alone should be reasonable here.

Reply:
we'd need to make number up since power factor is not 100% reliable indicator of peak. THD is better, but PTD doesn't gather that. so instead i opted for modifying the function and also grabbing power